examples/dmd/de_dg_advection_greedy.cpp

```diff
@@ -0,0 +1,1621 @@
+    // libROM parameterization parameters.
+    args.AddOption(&f_factor, "-ff", "--f-factor",
+                   "Frequency scalar factor.");
+    // libROM greedy parameters.
+    args.AddOption(&build_database, "-build_database", "--build_database",
+                   "-no-build_database", "--no-build_database",
+                   "Enable or disable the build_database phase.");
+    args.AddOption(&closest_rbf_val, "-crv", "--crv",
+                   "DMD Closest RBF Value.");
+    args.AddOption(&greedy_param_space_f_factor_max, "-greedy-param-f-factor-max",
+                   "--greedy-param-f-factor-max",
+                   "The maximum f_factor value of the parameter point space.");
+    args.AddOption(&greedy_param_space_f_factor_min, "-greedy-param-f-factor-min",
+                   "--greedy-param-f-factor-min",
+                   "The minimum f_factor value of the parameter point space.");
+    args.AddOption(&greedy_param_space_size, "-greedy-param-size",
+                   "--greedy-param-size",
+                   "The number of values to search in the parameter point space.");
+    args.AddOption(&greedy_relative_diff_tol, "-greedyreldifftol",
+                   "--greedyreldifftol", "The greedy algorithm relative diff tolerance.");
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+    // Differential Evolution Parameters
+    args.AddOption(&de, "-de", "--de",
+                   "-no-de", "--no-de",
+                   "Enable or disable the differential evolution phase.");
+    args.AddOption(&de_min_f_factor, "-de_min_ff", "--de_min_f-factor",
+                   "DE min f-factor");
+    args.AddOption(&de_max_f_factor, "-de_max_ff", "--de_max_f-factor",
+                   "DE max f-factor");
+    args.AddOption(&de_F, "-de_f", "--de_f",
+                   "DE F.");
+    args.AddOption(&de_CR, "-de_cr", "--de_cr",
+                   "DE CR.");
+    args.AddOption(&de_PS, "-de_ps", "--de_ps",
+                   "DE Population size.");
+    args.AddOption(&de_min_iter, "-de_min_iter", "--de_min_iter",
+                   "DE Minimum number of iterations.");
+    args.AddOption(&de_max_iter, "-de_max_iter", "--de_max_iter",
+                   "DE Maximum number of iterations.");
+    args.AddOption(&de_ct, "-de_ct", "--de_ct",
+                   "DE Cost threshold.");
+    args.Parse();
+    if (!args.Good())
+    {
+        if (mpi.Root())
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (mpi.Root())
+    {
+        args.PrintOptions(cout);
+    }
+
+    Device device(device_config);
+    if (mpi.Root()) {
+        device.Print();
+    }
+
+    io_dir = temp_io_dir;
```

Comment:
Running the example will crash if the io directory does not exist. We should probably check and throw an error, or just create it for the user:
```
std::filesystem::create_directory(io_dir);
```
You will need to add `#include <filesystem>` at the top as well.